addons/diagnostic/fnc_projectileTracking_drawProjectilePaths.sqf

```diff
@@ -0,0 +1,62 @@
+/* ----------------------------------------------------------------------------
+Internal Function: CBA_diagnostic_fnc_projectileTracking_drawProjectilePaths
+
+Description:
+    Draws the tracked projectiles flight paths
+
+Parameters:
+    _args - PerFrameHandler arguments <ARRAY>
+        _projectile - the projectile that needs to be tracked <OBJECT>
+        _index      - the current index/key of the hash <NUMBER>
+    _handle - PerFrameHandler handle <NUMBER>
+
+Returns:
+    nil
+
+Examples:
+    (begin example)
+        [CBA_diagnostic_fnc_projectileTracking_drawProjectilePaths, 0, []] call CBA_fnc_addPerFrameHandler;
+    (end)
+
+Author:
+    bux578
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+
+private _maxLines = GVAR(projectileMaxLines) min 20;
+
+for "_index" from 0 to _maxLines do {
+
+    if ( ([GVAR(projectileData), _index] call CBA_fnc_hashHasKey) ) then {
```

Comment:
if we are going for performance, we are doing an `in` and a `find` with `hashHasKey` and `hashGet`. We can probably skip one of them and just do a `hashGet` followed by a `!isNil` check.